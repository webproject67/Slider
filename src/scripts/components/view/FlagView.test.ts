import FlagView from './FlagView';

const state = {
  min: 0,
  max: 100,
  from: 0,
  fromPercent: 0,
  to: 100,
  toPercent: 100,
  step: 1,
  view: false,
  range: false,
  flag: true,
  progress: true,
  scale: true,
};

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new FlagView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    state.view = true;
    state.from = 1;
    state.fromPercent = 1;
    state.to = 99;
    state.toPercent = 99;
    expect(new FlagView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element3', () => {
    state.range = true;
    expect(new FlagView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element4', () => {
    state.view = false;
    expect(new FlagView(state).getElement()).toMatchSnapshot();
  });
});
