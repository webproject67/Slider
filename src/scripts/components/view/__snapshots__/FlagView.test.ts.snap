// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__pins"
>
  <div
    class="slider__pin slider__pin_position_maximum"
    style="left: 100%;"
  >
    100
  </div>
  <div
    class=""
  />
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__pins"
>
  <div
    class="slider__pin-vertical slider__pin-vertical_position_maximum"
    style="top: 99%;"
  >
    99
  </div>
  <div
    class=""
  />
</div>
`;

exports[`snapshot renders correctly element3 1`] = `
<div
  class="slider__pins"
>
  <div
    class="slider__pin-vertical slider__pin-vertical_position_minimum"
    style="top: 1%;"
  >
    1
  </div>
  <div
    class="slider__pin-vertical slider__pin-vertical_position_maximum"
    style="top: 99%;"
  >
    99
  </div>
</div>
`;

exports[`snapshot renders correctly element4 1`] = `
<div
  class="slider__pins"
>
  <div
    class="slider__pin slider__pin_position_minimum"
    style="left: 1%;"
  >
    1
  </div>
  <div
    class="slider__pin slider__pin_position_maximum"
    style="left: 99%;"
  >
    99
  </div>
</div>
`;
